src/tests/cipher.c

```diff
@@ -633,20 +633,29 @@ sm2_roundtrip(void **state)
     const pgp_ecc_pubkey_t *pub_ecc = &pub_key->key.ecc;
     const pgp_ecc_seckey_t *sec_ecc = &sec_key->key.ecc;
 
-    size_t        ctext_size = sizeof(ctext_buf);
-    pgp_errcode_t enc_result =
-      pgp_sm2_encrypt(ctext_buf, &ctext_size, key, sizeof(key), pub_ecc);
-    rnp_assert_int_equal(rstate, enc_result, PGP_E_OK);
+    uint8_t hashes[] = {PGP_HASH_SM3, PGP_HASH_SHA256, PGP_HASH_SHA512, 0};
```

Comment:
Maybe instead of the sentinel, use `ARRAY_SIZE` below (since we do that in some other places). Regardless, LGTM.

Reply:
Fixed